### PR TITLE
Implement keep filter to preserve matching images

### PR DESCRIPTION
**Pull request by NoUseFreak**
If provided, the filter will prevent matches from being deleted.

### CODE REVIEW

**Comment by NoUseFreak** [comment]
@danielfm Anything I need to change?

File: README.md [hunk 1]
```diff
@@ -68,6 +68,8 @@ Usage of ./bin/kube-ecr-cleanup-controller:
     	just log, don't delete any images.
   -interval int
     	check interval, in minutes. (default 30)
+  -keep-filters string
+        specify a filter or regex that when matched will preserve the matching images.
```
Comment (danielfm) [inline README.md]: The description here doesn't match the code.

```suggestion
        comma-separated list of filters or regexes that when matched will preserve the matching images.
```

**Review by danielfm — APPROVED** [review]
LGTM, just a nit.

**Comment by danielfm** [comment]
Thanks for your contribution! 🎉

File: pkg/processor/processor.go [hunk 2]
```diff
@@ -76,17 +76,20 @@ func RemoveOldImages(t *core.CleanupTask, kubeClient kubernetes.KubernetesClient
 		glog.V(10).Infof("Max Images is %d", t.MaxImages)
 		unusedOldImages := aws.FilterOldUnusedImages(t.MaxImages, images, usedImages[repoName])
 
-		if len(unusedOldImages) == 0 {
+		unusedImages := utils.ApplyKeepFilters(unusedOldImages, t.KeepFilters)
+		glog.Infof("Number of images after blacklist filter: %d", len(unusedImages))
```
Comment (sudermanjr) [inline pkg/processor/processor.go]: This is the only place it is ever called a blacklist filter which is rather confusing.

**Pull request merged**